### DISCUSSION TITLE
[Question] add/alter controls and update from different threads?

### DISCUSSION THREAD

**Discussion by gbrandt** [flet-dev/flet#1450] [Q&A]
### Question

Is it ok to add controls or edit controls on a page from a thread and call update from that thread? From multiple threads at the same time? Or does Flet get confused.  I’ve been doing it with some success but have recently gotten strange errors like a button not being clickable even though it has an on click handler. 

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by hoonlight** [comment]
> Threading
> 
> Technically, nobody will stop you from using threading library in async app, but it would be a bad idea. asyncio versions of locks, queues and tasks, used by Flet API are not thread-safe and, for example, calling await page.update_async() from multiple threads will lead to unpredictable results. Also, threading library is not supported by Pyodide if you decide to [deploy your app as a static website](https://flet.dev/docs/guides/python/publishing-static-website).
> 
> To run something in the background use [asyncio.create_task()](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task). For example, an async version of "countdown" control from [User controls](https://flet.dev/docs/guides/python/user-controls) guide would be:

https://flet.dev/docs/guides/python/async-apps
